fixitfelix/fix.py

```diff
@@ -151,3 +152,85 @@ def export_correct_data(
                     write_chunks_to_file(
                         tdms_writer, index_ranges, group, channel
                     )
+
+
+def export_correct_data(
+    filename: str, meta: source.MetaData, output_file: str
+) -> None:
+    """ Accepts either a path to a tdms file or to a folder with just tdms files to correct.
+    The name of the resulting folder or file is defined by output_file.
+    If output_file is empty, the name of the resulting file or folder is the previous name with '_corrected' as suffix.
+    All files in a folder will retain their previous name with '_corrected' suffix.
+    In case of a folder as input all tdms files are first checked for consistency so no file is corrected before each file is checked.
+    Afterwards all files are corrected and exported. This prevents cases where a later file is not valid for correction.
+    If a single file is given, the file is checked and corrected immediately.
+
+    Arguments:
+    filename: Path to the tdms file or folder with tdms files to correct.
+    meta: MetaData dict that contains all information needed for correction.
+    output_file: File path for the corrected TDMS file or folder.
+    """
+
+
+    #Determines generalized export path
+
+    path = pathlib.Path(filename)
+
+    p = either.Right(path) | error_handling.check_input_path
+    if isinstance(p, either.Left):
+            raise Exception(error_handling.ERROR_DESCRIPTIONS.get(p._value))
+
+    if output_file == "":
+        name = path.with_suffix("").name + "_corrected"
+        export_path = path.parent.joinpath(name)
+    else:
+        export_path = pathlib.Path(output_file)
+
+    p = either.Right(export_path) | check_export_path
+    if isinstance(p, either.Left):
+        raise Exception(error_handling.ERROR_DESCRIPTIONS.get(p._value))
+
+
+    # Directory and single file are handled seperately
+
+    if path.is_dir():
+        p = either.Right(path) | error_handling.check_dir_empty
+        if isinstance(p, either.Left):
+            raise Exception(error_handling.ERROR_DESCRIPTIONS.get(p._value))
+
+        if export_path.exists():
+            shutil.rmtree(export_path)
+        os.mkdir(export_path)
+
+
+        # Checks each file in folder for consistency
+
+        files_in_dir = len([f for f in path.iterdir()])
+        file_count = 0
+        pre_files = []
+        for tdms_file in path.iterdir():
+            pre_data = {}
+            pre_data["tdms_file"] = tdms_file
+            file_count += 1
+            print(f"Preprocess file {file_count} of {files_in_dir} at {tdms_file}")
+            name = tdms_file.with_suffix("").name + "_corrected.tdms"
+            pre_data["export_file_path"] = export_path.joinpath(name)
+            pre_data["source_file"] = preprocess(meta=meta, path=tdms_file)
+            pre_files.append(pre_data)
```

Comment:
Is it reasonable to define a new datatype PreData here with the attributes above instead of a generic dictionary?
This could make the code a bit more specific here. What do you think?

Reply:
I don't see an immediate advantage, but for readability and some future functionality it could be helpful to be more specific here.